src/libraries/System.Security.Cryptography/tests/X509Certificates/CertificateCreation/PrivateKeyAssociationTests.cs

```diff
@@ -909,6 +942,134 @@ public static void CheckCopyWithPrivateKey_MLDsa_OtherMLDsa_PrivateSeed()
+                creationParameters.Parameters.Add(parameterSet);
+
+                cngKey = CngKey.Create(CngAlgorithm.MLDsa, KeyName, creationParameters);
+
+                using (MLDsaCng mldsaCng = new MLDsaCng(cngKey))
+                {
+                    CertificateRequest request = new CertificateRequest(
+                        new X500DistinguishedName($"CN={KeyName}"),
+                        mldsaCng);
+
+                    DateTimeOffset now = DateTimeOffset.UtcNow;
+
+                    using (X509Certificate2 cert = request.CreateSelfSigned(now, now.AddDays(1)))
+                    using (MLDsa mldsa = cert.GetMLDsaPrivateKey())
+                    {
+                        mldsa.SignData("test"u8, signature);
+                        Assert.True(mldsaCng.VerifyData("test"u8, signature));
+                    }
+                }
+
+                // Some certs have disposed, did they delete the key?
+                using (CngKey stillPersistedKey = CngKey.Open(KeyName, CngProvider.MicrosoftSoftwareKeyStorageProvider))
+                using (MLDsaCng mldsa = new MLDsaCng(stillPersistedKey))
+                {
+                    mldsa.SignData("test"u8, signature);
+                }
+            }
+            finally
+            {
+                cngKey?.Delete();
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void AssociateEphemeralKey_CNG_MLDsa()
+        {
+            CngKeyCreationParameters creationParameters = new CngKeyCreationParameters();
+            CngProperty parameterSet = MLDsaTestHelpers.GetCngProperty(MLDsaAlgorithm.MLDsa44);
+            creationParameters.Parameters.Add(parameterSet);
+
+            byte[] signature = new byte[MLDsaAlgorithm.MLDsa44.SignatureSizeInBytes];
+
+            using (CngKey ephemeralCngKey = CngKey.Create(CngAlgorithm.MLDsa, keyName: null, creationParameters))
+            {
+                using (MLDsaCng mldsaCng = new MLDsaCng(ephemeralCngKey))
+                {
+                    CertificateRequest request = new CertificateRequest(
+                        new X500DistinguishedName($"CN=EphemeralMLDsaKey"),
+                        mldsaCng);
+
+                    DateTimeOffset now = DateTimeOffset.UtcNow;
+
+                    using (X509Certificate2 cert = request.CreateSelfSigned(now, now.AddDays(1)))
+                    using (MLDsa mldsa = cert.GetMLDsaPrivateKey())
+                    {
+                        mldsa.SignData("test"u8, signature);
+                        Assert.True(mldsaCng.VerifyData("test"u8, signature));
+                    }
+                }
+
+                // Some certs have disposed, did they delete the key?
+                using (MLDsaCng mldsa = new MLDsaCng(ephemeralCngKey))
+                {
+                    mldsa.SignData("test"u8, signature);
```

Comment:
Based on my comments in CopyWithPrivateKey.... I feel like this should either be exploding... or there's a memory leak.